src/main/java/com/researchspace/service/impl/RecordManagerImpl.java

```diff
@@ -1158,25 +1148,19 @@ private Folder getFolderOrRecordParent(Long parentId, User user, Record original
    * Retrieve all viewable documents by specific user (Role).
    *
    * @param subject the user whose viewable records we are retrieving
-   * @param onlyTemplates
-   * @return
    */
   private List<BaseRecord> getViewableRecordsByRole(User subject, boolean onlyTemplates) {
-    List<BaseRecord> result = new ArrayList<>();
-
+    Set<BaseRecord> viewableRecords = new HashSet<>();
```

Comment:
not sure what we do with this collection but it looks like that a `LinkedHashSet` scanning a set would be slightly more efficient here

Reply:
I don't think there will be any benefit of using a `LinkedHashSet` here. Any performance benefit in iteration is probably balanced out by the extra time taken for adding new elements and maintaining the order. Also we don't really iterate over this list any more either, and it gets converted back to an arraylist as the return.